docs/sphinx/using/examples/building_kernels.rst

```diff
@@ -1,9 +1,263 @@
 Building Kernels
--------------------------
+================
 This section will cover the most basic CUDA-Q construct, a quantum kernel.
 Topics include, building kernels, initializing states, and applying gate operations.
 
+Defining Kernels
+----------------
+
+Kernels are the building blocks of quantum algorithms in CUDA-Q. A kernel is specified by using the following syntax. `cudaq.qubit` builds a register consisting of a single qubit, while `cudaq.qvector` builds a register of :math:`N` qubits.
+
+.. tab:: Python
+
+   .. literalinclude:: ../../examples/python/building_kernels.py
+      :language: python
+      :start-after: [Begin Definition]
+      :end-before: [End Definition]
+
+.. tab:: C++
+
+   .. literalinclude:: ../../examples/cpp/building_kernels.cpp
+      :language: cpp
+      :start-after: [Begin Definition]
+      :end-before: [End Definition]
+
+Inputs to kernels are defined by specifying a parameter in the kernel definition along with the appropriate type. The kernel below takes an integer to define a register of N qubits.
+
+.. tab:: Python
+
+   .. literalinclude:: ../../examples/python/building_kernels.py
+      :language: python
+      :start-after: [Begin `InputDefinition`]
+      :end-before: [End `InputDefinition`]
+
+.. tab:: C++
+
+   .. literalinclude:: ../../examples/cpp/building_kernels.cpp
+      :start-after: [Begin `InputDefinition`]
+      :end-before: [End `InputDefinition`]
+
+Initializing states
+-------------------
+
+It is often helpful to define an initial state for a kernel. There are a few ways to do this in CUDA-Q. Note, method 5 is particularly useful for cases where the state of one kernel is passed into a second kernel to prepare its initial state.
```

Comment:
Please share these examples with Anna - it would be good to double check that we can compile them also for quantum hardware.